Restructure documentation in test-infra (#110)
kyma-project/test-infra: docs/prow/prow-installation-on-forks.md

- [gcloud](https://cloud.google.com/sdk/gcloud/) to communicate with Google Cloud Platform.
- OpenSSL

## Provision a cluster

1. Export these variables:

```

export PROJECT={project-name}
export CLUSTER_NAME={cluster-name}
export ZONE={zone-name}

```

2. When you communicate for the first time with the Google Cloud, set the context to your Google Cloud project. Execute this command:

```
gcloud config set project $PROJECT
```

3. Run the [`provision-cluster.sh`](../../development/provision-cluster.sh) script or follow [this](https://github.com/kubernetes/test-infra/blob/master/prow/getting_started_deploy.md#create-the-cluster) instruction to provision a new cluster on GKE. Ensure that kubectl points to the correct cluster. For GKE, execute the following command:

```
gcloud container clusters get-credentials $CLUSTER_NAME --zone=$ZONE --project=$PROJECT
```

## Create a bot account

Create a separate GitHub account which serves as a bot account that triggers the Prow comments that you enter in the pull request. If the Prow bot account is the same as the account that creates a job-triggering comment, the job is not triggered.

> Add the bot account to the [collaborators](https://help.github.com/articles/adding-outside-collaborators-to-repositories-in-your-organization/) on your personal GitHub account.

Reviewer: on your GitHub repository/fork

1. Create two buckets on GCS, one for storing secrets and the second for storing logs.

>**NOTE:** The bucket for storing logs is used in Prow by the Plank component. This reference is defined in the `config.yaml` file.

2. Create the following service accounts, role bindings, and private keys, encrypt them using Key Management Service (KMS), and upload them to your Secret storage bucket:

- **sa-gke-kyma-integration** with roles that allow the account to create Kubernetes clusters:
    - Kubernetes Engine Cluster Admin (`roles/container.clusterAdmin`)
    - Service Account User (`roles/iam.serviceAccountUser`)
- **sa-vm-kyma-integration** with roles that allow the account to provision virtual machines:
    - Compute Instance Admin (beta) (`roles/compute.instanceAdmin`)
    - Compute OS Admin Login (`roles/compute.osAdminLogin`)
    - Service Account User (`roles/iam.serviceAccountUser`)
- **sa-gcs-plank** with the role that allows the account to store objects in a Bucket:
    - Storage Object Admin (`roles/storage.objectAdmin`)
- **sa-gcr-push** with the role that allows the account to push images to Google Container Repository:
    - Storage Admin `roles/storage.admin`

> See an example of variables you need to export for each account:

Reviewer: I think that example should be moved to `prow-secrets-management.md` instruction. These variables are not used in scripts used in this instruction.

```bash
./install-prow.sh
```

> **NOTE:** The scripts prompts you to enter your OAuth2 token.

This script performs the following steps to install Prow:

- Deploy the NGINX Ingress Controller.
- Create a ClusterRoleBinding.
- Create a HMAC token to be used for GitHub Webhooks.
- Create secrets for HMAC and OAuth2 to be used by Prow.
- Deploy Prow components using the `starter.yaml` file from the `prow/cluster` directory.
- Add annotations for the Prow Ingress to make it work with the NGINX Ingress Controller.

## Verify the Installation

Verify if Prow installed successfully.

1. Check if all Pods are up and running:

   ```
   kubectl get pods
   ```

2. Check if the Deck is accessible from outside of the cluster:

   ```
   kubectl get ingress ing
   ```

   Copy the address of the ingress `ing` and open it in a browser to display the Prow status on the dashboard.

## Configure the webhook

After Prow installs successfully, you need to:

1. [Configure a webhook](https://support.hockeyapp.net/kb/third-party-bug-trackers-services-and-webhooks/how-to-set-up-a-webhook-in-github) to enable sending Events from a GitHub repository to Prow.

> 2. Add the bot account to [collaborators](https://help.github.com/articles/adding-outside-collaborators-to-repositories-in-your-organization/) on your forked repository and set it with push access rights. The bot account needs to accept your invitation.

Reviewer: Hm... similar point is in `## Create a bot account` section.

Author: Yup, I fixed that. Now it is mentioned only once ;)

## Configure Prow

When you use the [`install-prow.sh`](../../development/provision-cluster.sh) script to install Prow on your cluster, the list of plugins and configuration is empty. You can configure Prow by specifying the `config.yaml` and `plugins.yaml` files.

### The config.yaml file

> The `config.yaml` file contains the basic Prow configuration. When you create a particular Prow job, it uses the Pod Preset definitions from this file. See the example of such a file [here](../../development/config.yaml).

Reviewer: ```suggestion
The `config.yaml` file contains the basic Prow configuration. When you create a particular Prow job, it uses the Pod Preset definitions from this file. See the example of such a file [here](../../prow/config.yaml).
```

For more details, see the [Kubernetes documentation](https://github.com/kubernetes/test-infra/blob/master/prow/getting_started_deploy.md#add-more-jobs-by-modifying-configyaml).

### The plugins.yaml file

>  The `plugins.yaml` file contains the list of [plugins](https://prow.k8s.io/plugins) you enable on a given repository. See the example of such a file [here](../../development/plugins.yaml).

Reviewer: ```suggestion
 The `plugins.yaml` file contains the list of [plugins](https://status.build.kyma-project.io/plugins) you enable on a given repository. See the example of such a file [here](../../prow/plugins.yaml).
```

For more details, see the [Kubernetes documentation](https://github.com/kubernetes/test-infra/blob/master/prow/getting_started_deploy.md#enable-some-plugins-by-modifying-pluginsyaml).

### Verify the configuration

> To check if the `plugins.yaml` and `config.yaml` configuration files are correct, run the `check.sh {file_path}` script. For example, run:

Reviewer: ```suggestion
To check if the `plugins.yaml` and `config.yaml` configuration files are correct, run the `check.sh {plugins_file_path} {config_file_path}` script. For example, run:
```
